### DISCUSSION TITLE
Species filter

### DISCUSSION THREAD

**Discussion by 1gkwc6** [sheltermanager/asm3#1549] [Q&A / Help] [answered]
Is there any way of filtering user accounts down to a specific species. 

The Shelter view, Home page (Recently Entered Shelter) and Calendar views appear very busy and cluttered for our cattery users. Our cattery is about on fifth the size of the kennels.

**Comment by bobintetley** [comment]
You can't filter by species, but you can put a location filter on your cattery users so that they can only see animals in cattery locations on those screens.

To do that, turn on "Enable location filters" under Settings->Options->Display to add a new location filter dropdown to the user accounts screen.

**Reply by 1gkwc6** [reply]
This works until it comes to foster animals. When I add Fostered Animals to location filter, they show up on Shelter View and the Medical Calendar. 
I was thinking creating x2 animal flags. Foster Dog and Foster Cat, might be a solution, but will volunteers remember to add the flag after they have selected the Movement:: Movement type::Foster ? Probably not. 
Could I put a request in to have the Movement type developed for our organisation to split the Movement Type::Foster  into ::Foster Dog and ::Foster Cat?

**Reply by bobintetley** [reply]
We would not add new movement types, that doesn't make sense when it is a combination of movement and species.

It sounds like what you want is a location filter for foster animals of a particular species.

**Reply by gitcodehelp** [reply]
I second this request for a direct Species filter.

Our shelter has separate medical and adoption staff for dogs and cats, and none of them want the other species cluttered on their screens.  Because the location filter selection is inclusive instead of exclusive, it is cumbersome as an administrator to assign a dog user group "every location except the cat locations", etc.

This is compounded by the limitation of having only 2 levels of location refinement (location + unit).  For example, our shelter has a separate cat building with several rooms (isolation, kitten, adult, etc.), each with their own units.  To give cat staff the location refinement they want, each room must be defined as a separate location (Cat Bldg-Iso, Cat Bldg-Room A, ...), which makes for a long list of items to include/exclude in a user filter.  A sub-level of location refinement for rooms would be very helpful.

A location filter for fosters of a particular species would definitely be helpful in the interim.

**Reply by bobintetley** [reply, marked as answer]
We fairly recently refactored the location filter code to make it easier to add new ones in future. Adding a species filter has turned out to be fairly trivial as a result. I've added 2 new location filter options for "All Dogs" and "All Cats" to the list. They've gone in this morning's update and allow you to blanket assign all cats in care or all dogs in care as a single filter.
